### ISSUE TITLE
azan is not playing in android build 

### ISSUE THREAD

**Issue by umerhayat54** [meypod/al-azan#25]

**Comment by meypod** [comment]
Can you provide more info ?

**Comment by meypod** [comment]
Please try latest version if possible:
https://github.com/meypod/al-azan/releases/download/v1.10.0-rc.29/al-azan--release-1.10.0-rc.29--universal-38000-signed.apk

**Comment by umerhayat54** [comment]
yes, I install it in apk but azan not working. battery permission disable also

**Comment by umerhayat54** [comment]
okay lets me check APK also

**Comment by umerhayat54** [comment]
yes it working Thanks bro

**Comment by meypod** [comment]
I'm glad it's working

**Comment by umerhayat54** [comment]
Hi, @meypod  I hope you are well. I am facing an issue to run your project on ios. please can you guide me or if you can provide ios IPA it will be good for me Thanks

**Comment by meypod** [comment]
Hi
I'm sorry, I know this is a react native app, but currently app is not made to work on ios
Maybe in future when I have access to apple products I'll work on it, but for now it's not going to happen

**Comment by umerhayat54** [comment]
@meypod okay thanks

**Comment by umerhayat54** [comment]
@meypod Hi, I hope you are well. 
do you try to implement it on ios? 
Thanks

**Comment by umerhayat54** [comment]
> Hi @umerhayat54 I am sorry but I do not have apple products, so I can't really work on it

Thanks For your reply.  it's okay.  can you share if you have to do what will you need to change in it?

**Comment by meypod** [comment]
**tl;dr;** there's a lot of work
-----------------------

There's a lot of things that needs testing, but in general these are the things I can remember at the moment that certainly need work:
1. the most important feature, which is relied on, is the alarm manager of android in notifee library. have to check how it works on iOS, because in android I'm using it to launch the dismiss adhan page.
2. then it's media player. for good reasons that I don't remember exactly now (probably the fine control over mid-call behavior and other stuff), I decided to write the media player using android's native media player. for iOS either it has to be done manually like the android part or it has to be swapped out with a cross platform solution.
3. both native widget and notification widget rely on android's native xml layouts. so they probably need to be created from scratch for iOS as well
4. qibla compass module and it's magnetic pole adjustment needs to be redone in iOS
5. after everything is done, should hide the android specific settings (like common problems page) from user

**Comment by umerhayat54** [comment]
> ## **tl;dr;** there's a lot of work
> There's a lot of things that needs testing, but in general these are the things I can remember at the moment that certainly need work:
> 
> 1. the most important feature, which is relied on, is the alarm manager of android in notifee library. have to check how it works on iOS, because in android I'm using it to launch the dismiss adhan page.
> 2. then it's media player. for good reasons that I don't remember exactly now (probably the fine control over mid-call behavior and other stuff), I decided to write the media player using android's native media player. for iOS either it has to be done manually like the android part or it has to be swapped out with a cross platform solution.
> 3. both native widget and notification widget rely on android's native xml layouts. so they probably need to be created from scratch for iOS as well
> 4. qibla compass module and it's magnetic pole adjustment needs to be redone in iOS
> 5. after everything is done, should hide the android specific settings (like common problems page) from user
Thanks for sharing.

**Comment by umerhayat54** [comment]
> ## **tl;dr;** there's a lot of work
> There's a lot of things that needs testing, but in general these are the things I can remember at the moment that certainly need work:
> 
> 1. the most important feature, which is relied on, is the alarm manager of android in notifee library. have to check how it works on iOS, because in android I'm using it to launch the dismiss adhan page.
> 2. then it's media player. for good reasons that I don't remember exactly now (probably the fine control over mid-call behavior and other stuff), I decided to write the media player using android's native media player. for iOS either it has to be done manually like the android part or it has to be swapped out with a cross platform solution.
> 3. both native widget and notification widget rely on android's native xml layouts. so they probably need to be created from scratch for iOS as well
> 4. qibla compass module and it's magnetic pole adjustment needs to be redone in iOS
> 5. after everything is done, should hide the android specific settings (like common problems page) from user

Thank for sharing.

**Comment by umerhayat54** [comment]
@meypod  can you please tell how you handle Azan notifications when the app is in a kill state? Thanks

**Comment by meypod** [comment]
@umerhayat54 as far as I understand, you can't do anything if app is killed (all scheduled alarms are wiped, etc)
however if the app is swiped away (not killed explicitly), you have to look at what https://github.com/invertase/notifee/ does, because that's what I use for scheduling and handling notifications

in the second case (if the app is not killed explicitly and is sitting in the background), if that's what you are asking for, since I have already look at the notifee's code, what it does is it keeps and reuses application context everywhere in the code 

like when trying to run a headless task from a broadcast, it reuses the broadcast context for running the task
this prevents some common issues and restrictions that are set on background processes :) 

I hope this helps

**Comment by umerhayat54** [comment]
> @umerhayat54 as far as I understand, you can't do anything if app is killed (all scheduled alarms are wiped, etc) however if the app is swiped away (not killed explicitly), you have to look at what https://github.com/invertase/notifee/ does, because that's what I use for scheduling and handling notifications
> 
> in the second case (if the app is not killed explicitly and is sitting in the background), if that's what you are asking for, since I have already look at the notifee's code, what it does is it keeps and reuses application context everywhere in the code
> 
> like when trying to run a headless task from a broadcast, it reuses the broadcast context for running the task this prevents some common issues and restrictions that are set on background processes :)
> 
> I hope this helps

yes it is helpful for me Thanks let's me see it
